### ISSUE TITLE
documentation - autodoc - document configable values

### ISSUE THREAD

**Issue by vEpiphyte** [vertexproject/synapse#338]
Update autodoc to walk the synapse package and instantiate any objects which inherit the Configable mixin, and generate docstrings for the configable values of that object.  We should inspect whether or not a object has runtime-setable values and note those as well (this can be done by inspecting the event bus's _syn_funcs for keys starting with syn:conf:set).  This should generate a single .rst file documenting configable objects.

We should be smart so that we're not duplicating config options in the case of inheritance.  We should document the inheritance instead and provide intra-doc links to the inherited objects.

Objects which inherit from configable should have a docstring which points to the autogenerated doc for easy reference.

**Comment by invisig0th** [comment]
Maybe we could update the convention to be a staticmethod that returns the config defs for a given object, and have autodoc generate them based on that?  I could see some crazyness with having to instantiate an object in order to gen docs about it's config options...

**Comment by vEpiphyte** [comment]
@invisig0th that sounds like a sane approach; and is something we can do once in configable.

**Comment by vEpiphyte** [comment]
Actually @invisig0th that would only work if we moved all the conf-defs for configable synapse objects to reside in that static method; which also risks one class from stomping the 'configable' opts of another class.

**Comment by vEpiphyte** [comment]
@invisig0th doing this with specially decorated static methods gives us a quick and easy 90% solution (we still dont know if a method is runtime-setable not).  See https://github.com/vEpiphyte/synapse/tree/epiphyte_autodoc_338 for a POC of that update to config.  It would look like the following:

```
import synapse.lib.config as s_config
class Foo(s_config.Config):

    @staticmethod
    @s_config.confdef()
    def foodefs():
        defs = (
            ('fooval', {'type': 'int', 'doc': 'what is foo val?', 'defval': 99}),
            ('enabled', {'type': 'bool', 'doc': 'is thing enabled?', 'defval': 0}),
        )
        return defs
```

this allows the opts retrieval without needing to make class instances and the configable __init__ will automatically find and load the functions.

**Comment by vEpiphyte** [comment]
Closed by #362
